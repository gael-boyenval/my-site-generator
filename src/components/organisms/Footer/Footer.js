import React, { PureComponent } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { colors } from 'Styles/tokens';
import { SubTitle, TinyHeading, BodySans, Mention } from 'Atoms/Typography';
import Container from 'Layout/Container';
import { Padding, Margin } from 'Layout/Spaces';

const Wrapper = styled.footer`
  text-align: center;
`;

const SocialLink = TinyHeading.withComponent('a').extend`
  &:not(:last-child):after {
    content:" - "
  }
`;

const EmailButton = styled.button`
  border-bottom: solid 1px currentColor;
  transition: color 0.2s ease;

  &:hover {
    color: ${colors.electric};
  }
`;

const BodySansDiv = BodySans.withComponent('div');

class Footer extends PureComponent {
  static propTypes = {
    social: PropTypes.shape({}).isRequired,
  };

  handleContact = () => {
    window.location.assign('mailto:gaelboyenval@icloud.com');
  };

  render() {
    const { social } = this.props;

    return (
      <Wrapper>
        <Container>
          <Padding vertical="wide">
            <Margin bottom="m">
              <SubTitle>Prenons un café</SubTitle>
            </Margin>
            <Margin bottom="xs">
              {Object.keys(social).map((netWork) => (
                <SocialLink key={netWork} href={social[netWork]} target="_blank">
                  {netWork}
                </SocialLink>
              ))}
            </Margin>
            <Margin bottom="l">
              <BodySansDiv>
                <Margin bottom="xs">
                  <EmailButton onClick={this.handleContact}>Contactez-moi par e-mail</EmailButton>
                </Margin>
                06 08 14 58 40
              </BodySansDiv>
            </Margin>
            <Mention>
              Ce site à été créé avec beaucoup d’amour, <br />
              mais aussi avec Gatsby, React et est hebergé par Github
            </Mention>
          </Padding>
        </Container>
      </Wrapper>
    );
  }
}

export default Footer;
