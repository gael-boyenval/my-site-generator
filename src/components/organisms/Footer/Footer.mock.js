export default {
  social: {
    github: 'https://github.com/gael-boyenval',
    linkedin: 'https://www.linkedin.com/in/gaÃ«l-boyenval-5b931415/',
    medium: 'https://medium.com/@gael_boyenval',
    twitter: 'https://twitter.com/gael_boyenval',
  },
};
