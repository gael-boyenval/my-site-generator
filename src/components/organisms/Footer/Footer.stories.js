/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Footer from './Footer';
import mock from './Footer.mock';

storiesOf('04 - Organisms', module).add('Footer', withInfo()(() => <Footer {...mock} />));
