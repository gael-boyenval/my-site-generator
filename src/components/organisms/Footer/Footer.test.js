/* eslint-disable no-undef */
import React from 'react';
import Footer from './Footer';
import mock from './Footer.mock';

describe('Component Footer', () => {
  it('should be rendered without crash', () => {
    const component = shallow(<Footer {...mock} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call handleContact when button is clicked', () => {
    const component = mount(<Footer {...mock} />);
    window.location.assign = jest.fn();
    component.find('button').simulate('click');
    expect(window.location.assign).toBeCalledWith('mailto:gaelboyenval@icloud.com');
  });
});
