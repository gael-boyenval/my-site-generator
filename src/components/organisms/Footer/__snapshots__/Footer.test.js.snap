// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Footer should be rendered without crash 1`] = `
<Footer__Wrapper>
  <Container>
    <Spaces__Padding
      all={false}
      bottom={false}
      horizontal={false}
      inline={false}
      left={false}
      right={false}
      top={false}
      vertical="wide"
    >
      <Spaces__Margin
        all={false}
        bottom="m"
        horizontal={false}
        inline={false}
        left={false}
        right={false}
        top={false}
        vertical={false}
      >
        <Typography__SubTitle>
          Prenons un café
        </Typography__SubTitle>
      </Spaces__Margin>
      <Spaces__Margin
        all={false}
        bottom="xs"
        horizontal={false}
        inline={false}
        left={false}
        right={false}
        top={false}
        vertical={false}
      >
        <Typography__TinyHeading
          href="https://github.com/gael-boyenval"
          key="github"
          target="_blank"
        >
          github
        </Typography__TinyHeading>
        <Typography__TinyHeading
          href="https://www.linkedin.com/in/gaël-boyenval-5b931415/"
          key="linkedin"
          target="_blank"
        >
          linkedin
        </Typography__TinyHeading>
        <Typography__TinyHeading
          href="https://medium.com/@gael_boyenval"
          key="medium"
          target="_blank"
        >
          medium
        </Typography__TinyHeading>
        <Typography__TinyHeading
          href="https://twitter.com/gael_boyenval"
          key="twitter"
          target="_blank"
        >
          twitter
        </Typography__TinyHeading>
      </Spaces__Margin>
      <Spaces__Margin
        all={false}
        bottom="l"
        horizontal={false}
        inline={false}
        left={false}
        right={false}
        top={false}
        vertical={false}
      >
        <Typography__BodySans>
          <Spaces__Margin
            all={false}
            bottom="xs"
            horizontal={false}
            inline={false}
            left={false}
            right={false}
            top={false}
            vertical={false}
          >
            <Footer__EmailButton
              onClick={[Function]}
            >
              Contactez-moi par e-mail
            </Footer__EmailButton>
          </Spaces__Margin>
          06 08 14 58 40
        </Typography__BodySans>
      </Spaces__Margin>
      <Typography__BodySans>
        Ce site à été créé avec beaucoup d’amour, 
        <br />
        mais aussi avec Gatsby, React et est hebergé par Github
      </Typography__BodySans>
    </Spaces__Padding>
  </Container>
</Footer__Wrapper>
`;
